lib/class-wp-block-templates-registry.php

```diff
@@ -0,0 +1,272 @@
+<?php
+/**
+ * Block template functions.
+ *
+ * @package Gutenberg
+ * @since 6.7.0
+ */
+
+if ( ! class_exists( 'WP_Block_Templates_Registry' ) ) {
+	/**
+	 * Core class used for interacting with block templates.
+	 *
+	 * @since 6.7.0
+	 */
+	final class WP_Block_Templates_Registry {
+		/**
+		 * Registered block templates, as `$name => $instance` pairs.
+		 *
+		 * @since 6.7.0
+		 * @var WP_Block_Template[] $registered_block_templates Registered block templates.
+		 */
+		private $registered_block_templates = array();
```

Comment:
Probably better to rename remaining instance `block_templates` -> `templates`.

Reply:
Makes sense. I changed it in 08b740794bf2b792bcd0d63ab97fbe615cb4780a. I also replaced `WP_Block_Templates_Registry` with `WP_Templates_Registry` to keep it aligned as well, but happy to undo it if necessary.